mod/groups/start.php

```diff
@@ -496,34 +496,45 @@ function groups_create_event_listener($event, $object_type, $object) {
  * Return the write access for the current group if the user has write access to it.
  */
 function groups_write_acl_plugin_hook($hook, $entity_type, $returnvalue, $params) {
-	$page_owner = elgg_get_page_owner_entity();
-	$user_guid = $params['user_id'];
+	
+	if (empty($params) || !is_array($params)) {
```

Comment:
> Is it a good idea to silently stop executing the handler just because someone calls it the wrong way? After all the whole hook is useless if there is no user_guid whose permissions to check.

elgg_extract type hinting will trigger php warnings is not passed an array

> Only use case I can think of is checking if $params['user_id'] === 0 which means that for some reason the hook was called for an non-logged-in user.

this can trigger a php warning for unset array key